client/my-sites/theme/main.jsx

```diff
@@ -80,36 +88,12 @@ const ThemeSheet = React.createClass( {
 		return !! this.props.name;
 	},
 
-	hideSiteSelectorModal() {
-		this.setState( { selectedAction: null } );
-	},
-
 	onPrimaryClick() {
-		if ( ! this.props.isLoggedIn ) {
-			this.props.signup( this.props );
-		} else if ( this.props.active ) {
-			this.props.customize( this.props, this.props.selectedSite );
-		} else if ( this.props.price ) {
-			this.selectSiteAndDispatch( 'purchase' );
-		} else {
-			this.selectSiteAndDispatch( 'activate' );
-		}
+		this.props.defaultOption.action( this.props );
 	},
 
 	onPreviewButtonClick() {
```

Comment:
`onPrimaryClick()` and `onPreviewButtonClick()` are now the same and could be combined into one function.